lib/fs.js

```diff
@@ -648,6 +654,14 @@ fs.readSync = function(fd, buffer, offset, length, position) {
     offset = 0;
   }
 
+  if (length === 0) {
```

Comment:
How do you propose to solve this? @bnoordhuis asked me to move the checks to `lib/fs.js` to avoid some kernel edge case.

Thoughts?